multimodal/src/autogluon/multimodal/utils/onnx.py

```diff
@@ -0,0 +1,98 @@
+import logging
+from typing import Dict, List, Optional, Tuple, Union
+
+from torch import tensor
+
+from ..constants import AUTOMM, FEATURE_EXTRACTION
+
+logger = logging.getLogger(AUTOMM)
+
+
+def get_onnx_input(pipeline: str, config: Optional[Dict] = None):
+    """
+    Get information for a predictor to export its model in onnx format.
+
+    Parameters
+    ----------
+    pipeline
+        The predictor's pipeline.
+    config
+        The predictor's config.
+
+    Returns
+    -------
+    valid_input
+        The valid keys for the input batch.
+    dynamic_axes
+        By default the exported model will have the shapes of all input and output tensors
+        set to exactly match those given in ``args``. To specify axes of tensors as
+        dynamic (i.e. known only at run-time), set ``dynamic_axes`` to a dict with schema:
+            * KEY (str): an input or output name. Each name must also be provided in ``input_names`` or
+              ``output_names``.
+            * VALUE (dict or list): If a dict, keys are axis indices and values are axis names. If a
+              list, each element is an axis index.
+        See torch.onnx.export for more explanations.
+    default_onnx_path
+        The default path of the export onnx model.
+    default_batch
+        The default batch to help trace and export the model.
+
+    """
+    default_onnx_path = None
+    if pipeline == FEATURE_EXTRACTION:
+        valid_input = [
+            "hf_text_text_token_ids",
+            "hf_text_text_valid_length",
+            "hf_text_text_segment_ids",
+        ]
+        dynamic_axes = {
+            "hf_text_text_token_ids": {
+                0: "batch_size",
+                1: "sentence_length",
+            },
+            "hf_text_text_valid_length": {
+                0: "batch_size",
+            },
+            "hf_text_text_segment_ids": {
+                0: "batch_size",
+                1: "sentence_length",
+            },
+        }
+        if config:
+            default_onnx_path = config["model"]["hf_text"]["checkpoint_name"].replace("/", "_") + ".onnx"
+        default_batch = {
+            "hf_text_text_token_ids_column_sentence1": tensor(
```

Comment:
`hf_text_text_token_ids_column_sentence1 ` is not in `valid_input`. Will it be used in the onnx export?

Reply:
It's used in tracing but is treated as constants onnx's graph. It only contributes to column feature but we extract only feature, so it won't affect ONNX's output.

Reply:
It's used in a previous version. Removed.